packages/account/src/providers/memory-cache.test.ts

```diff
@@ -53,81 +50,97 @@ describe('Memory Cache', () => {
     const ttl = 1000;
     const expiresAt = Date.now() + ttl;
     const memCache = new MemoryCache(ttl);
+    const value = randomBytes(8);
 
-    expect(memCache.set(CACHE_ITEMS[0])).toBeGreaterThanOrEqual(expiresAt);
+    expect(memCache.set(value)).toBeGreaterThanOrEqual(expiresAt);
   });
 
   it('can get [valid key]', () => {
-    const KEY = CACHE_ITEMS[1];
+    const value = randomBytes(8);
     const memCache = new MemoryCache(100);
 
-    memCache.set(KEY);
+    memCache.set(value);
 
-    expect(memCache.get(KEY)).toEqual(KEY);
+    expect(memCache.get(value)).toEqual(value);
   });
 
   it('can get [valid key bytes like]', () => {
-    const KEY = CACHE_ITEMS[2];
+    const value = randomBytes(8);
     const memCache = new MemoryCache(100);
 
-    memCache.set(KEY);
+    memCache.set(value);
 
-    expect(memCache.get(KEY)).toEqual(KEY);
+    expect(memCache.get(value)).toEqual(value);
   });
 
   it('can get [valid key, expired content]', async () => {
-    const KEY = randomBytes(8);
+    const value = randomBytes(8);
     const memCache = new MemoryCache(1);
 
-    memCache.set(KEY);
+    memCache.set(value);
 
     await new Promise((resolve) => {
       setTimeout(resolve, 10);
     });
 
-    expect(memCache.get(KEY)).toEqual(undefined);
+    expect(memCache.get(value)).toEqual(undefined);
   });
 
   it('can get, disabling auto deletion [valid key, expired content]', async () => {
-    const KEY = randomBytes(8);
+    const value = randomBytes(8);
     const memCache = new MemoryCache(1);
 
-    memCache.set(KEY);
+    memCache.set(value);
 
     await new Promise((resolve) => {
       setTimeout(resolve, 10);
     });
 
-    expect(memCache.get(KEY, false)).toEqual(KEY);
+    expect(memCache.get(value, false)).toEqual(value);
   });
 
   it('can delete', () => {
-    const KEY = randomBytes(8);
+    const value = randomBytes(8);
     const memCache = new MemoryCache(100);
 
-    memCache.set(KEY);
-    memCache.del(KEY);
+    memCache.set(value);
+    memCache.del(value);
 
-    expect(memCache.get(KEY)).toEqual(undefined);
+    expect(memCache.get(value)).toEqual(undefined);
   });
 
   it('can get active [with data]', () => {
-    const EXPECTED: BytesLike[] = [CACHE_ITEMS[0], CACHE_ITEMS[1], CACHE_ITEMS[2]];
+    const value1 = randomBytes(8);
+    const value2 = randomBytes(8);
+    const value3 = randomBytes(8);
+    const EXPECTED: BytesLike[] = [value1, value2, value3];
+
     const memCache = new MemoryCache(100);
 
-    expect(memCache.getActiveData()).toStrictEqual(EXPECTED);
+    memCache.set(value1);
+    memCache.set(value2);
+    memCache.set(value3);
+
+    expect(memCache.getActiveData()).containSubset(EXPECTED);
```

Comment:
Why `containSubset` instead of `toStrictEqual`?

Reply:
@arboleya because it won't be strictly equal. The class `MemoryCache` uses a global cache and the test suite interacts with it multiple times.

Reply:
Considering that values 1-3 are `identical`, and let's say those three `.set()` calls occur between a 100ms interval, does it mean that `getActiveData()` will always return a single item?

Reply:
I believe `getActiveData` returns all cached data that has not expired.

Reply:
In this case, it should always be `[value1, value2, value3]`. For the `toStrictEqual()` assertion to fail, the interval between the instantiation and the test assertion needs to be greater than the cache TTL—in this case, it is 100ms.

If the above understanding is correct, the flakiness issue could be attributed to this; some tests executed in `<100ms` and others in `>=100`, and the latter would fail.

Having a greater `ttl` and keeping with `strictEqual()` assertions could work best because of the improved predictability. The `containSubset` creates some uncertainty; it is not obvious what the expected result is and why.

Let me know what you think.

```ts
/* 100% pseudo-code */

const value1 = randomBytes(8);
const value2 = randomBytes(8);
const value3 = randomBytes(8);

const allValues: BytesLike[] = [value1, value2, value];

const ttl = 500;
const timeStart = new Date().getTime();
const memCache = new MemoryCache(ttl);

memCache.set(value1);
memCache.set(value2);
memCache.set(value3);

const activeData = memCache.getActiveData();
const timeDiff = new Date().getTime() - timeStart;

expect(timeDiff).toBeLessThan(ttl); // test must finish before ttl
expect(activeData).toBeStrictEqual(allValues); // everything must be active
```

Reply:
https://github.com/FuelLabs/fuels-ts/pull/2079/commits/189bdb8d97f0a3301bfe182de5611abd25b68ae0

https://github.com/FuelLabs/fuels-ts/actions/runs/8720023700/job/23920681093?pr=2079#step:10:509

Like I said before, the `MemoryCache` class uses a [global cache](https://github.com/FuelLabs/fuels-ts/blob/124099b1a134607b8d12953d61ac7065c34fc7ed/packages/account/src/providers/memory-cache.ts#L11), the instance only controls how long the values set by it are going to live within the cache.

```ts
const old = randomValues(8)

const instance1 = new MemoryCache(2000)
instance1.set(old)

await new Promise((resolve) => {
  setTimeout(resolve, 100);
});

const new = randomValues(8)

const instance2 = new MemoryCache(100)
instance2.set(new)

instance2. getActiveData() // contains both [old, new]
```

`toBeStrictEqual` was the reason this test case was flaky in the first place.

There are other test cases setting values within the _global_ cache and by the time this specific test case uses `toStrictEqual` to validate only some specific values, the validation fails because there is still other active data within the cache.

Reply:
Got it; my bad.

Maybe a comment could make it more straightforward.

```ts
/*
  MemoryCache uses a global cache with values from
  several instances, all returned by `getActiveData()`.
  However, we only want to check the ones from this
  test, so we use `containSubset`.
*/
expect(memCache.getActiveData()).containSubset(EXPECTED);
```

Reply:
Done https://github.com/FuelLabs/fuels-ts/pull/2079/commits/5dc5dcf4d24642365203fdb35bfa4b4c0be4102d